src/System.Private.ServiceModel/tests/Common/Infrastructure/TestEventListener.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+
+namespace Infrastructure.Common
+{
+    /// <summary>Simple event listener</summary>
+    /// This class is based on https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Diagnostics/Tracing/TestEventListener.cs
+    public sealed class TestEventListener : EventListener
+    {
+        private Dictionary<string, bool> _targetSourceName = new Dictionary<string, bool>();
+        private readonly EventLevel _level;
+
+        public Action<EventWrittenEventArgs> _eventWritten;
```

Comment:
> public Action<EventWrittenEventArgs> _eventWritten [](start = 7, length = 51)

It would be better to make this a property, not a public field.  The WcfTestCase class is modifying this member directly